tgctoolbox/settings.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import os
+
+import yaml
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+BASE_FILES = [
+    os.path.join("configuration", "base.yaml"),
+    os.path.join("configuration", "local.yaml"),
+    os.path.join("configuration", "production.yaml"),
+]
+
+
+class Settings:
+    """Class to manage application settings."""
+
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+        if cls._instance is None:
+            cls._instance = super().__new__(cls)
+        return cls._instance
+
+    def __init__(self, config_files=None, reload=False):
+        """
+        Initialize Settings object.
+
+        Args:
+            config_files (list): List of YAML configuration files.
+            reload (bool): Whether to reload settings.
+        """
+        if not hasattr(self, "_initialized") or reload:
+            self._settings = {}
+            config_files = config_files or BASE_FILES
+            self._load_from_yaml(config_files)
+            self._load_from_env()
+            self._initialized = True
+
+    def _load_from_yaml(self, config_files):
+        """Load settings from YAML files."""
+        loaded_settings = {}
+        settings_names = []
+        if config_files:
+            for file in config_files:

```

Comment:
when running a program we will have also env variable called `APP_ENVIRONMENT` which will choose from which files we should load the settings. `base.yaml` is always initialized but dependendly on APP_ENVIRONMENT=`local`|`production` we should choose correct file

Reply:
Do you want this as env var or loaded from yaml? Also, should I terminate if the file local / production is not found? Currently, I throw warning only.

Reply:
`APP_ENVIRONMENT` will be a var loaded from var

Reply:
Added my proposition in [fdcbb2e](https://github.com/TwoGuysCoding/TGCToolbox/pull/5/commits/fdcbb2e5b1838cdc23c068157b3a38f8c3c8728e). Currently this allows you to choose base dir in which the yaml files are defined and loads from `local | production` based on `APP_ENVIRONMENT` setting, defaulting to `local`.

Reply:
Also add a warning message that if `APP_ENVIRONEMT` doesn't exist it load by default from `local`

Reply:
Added in [d0941e7](https://github.com/TwoGuysCoding/TGCToolbox/pull/5/commits/d0941e7bf4347195c02e3b2d311db743fca86cb1)